api/debug.go

```diff
@@ -68,3 +113,189 @@ func TraceStructLog(ctx *rpc.CallContext, db state.StateDB, header *types.Header
 	}
 	return tracer.StructLogs(), nil
 }
+
+// TraceCall lets you trace a given eth_call. It collects the structured logs
+// created during the execution of EVM if the given transaction was added on
+// top of the provided block and returns them as a JSON object.
+// If no transaction index is specified, the trace will be conducted on the state
+// after executing the specified block. However, if a transaction index is provided,
+// the trace will be conducted on the state after executing the specified transaction
+// within the specified block.
+func (s *PrivateDebugAPI) TraceCall(ctx *rpc.CallContext, args TransactionArgs, blockNrOrHash *vm.BlockNumberOrHash, config *TraceCallConfig) (interface{}, error) {
+	bNrOrHash := vm.BlockNumberOrHashWithNumber(rpc.PendingBlockNumber)
+	if blockNrOrHash != nil {
+		bNrOrHash = *blockNrOrHash
+	}
+	var (
+		statedb state.StateDB
+		header *types.Header
+		chaincfg *params.ChainConfig
+		evmFn evmfunc
+	)
+
+	if config != nil  && config.TxIndex != nil {
+		var err error
+		statedb, header, evmFn, chaincfg, err = s.stateAtTransaction(ctx, bNrOrHash, int(*config.TxIndex))
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		err := s.evmmgr.View(bNrOrHash, args.from, ctx, func(h *types.Header, sdb state.StateDB, getEVM func (state.StateDB, *vm.Config, common.Address, *big.Int) *vm.EVM, cfg *params.ChainConfig) error{
+			statedb = sdb
+			header = h
+			chaincfg = cfg
+			evmFn = getEVM
+			return nil
+		})
+		if err != nil {
+			switch err.(type) {
+			case codedError:
+			default:
+				err = evmError{err}
+			}
+			return nil, err
+		}
+	}
+
+	if config != nil && config.StateOverrides != nil {
+		if err := config.StateOverrides.Apply(statedb); err != nil {
+			return nil, err
+		}
+	}
+
+	gasCap := new(GasPool).AddGas(header.GasLimit)
+	if err := args.CallDefaults(gasCap.Gas(), header.BaseFee, chaincfg.ChainID); err != nil {
+		return nil, err
+	}
+	msg, err := args.ToMessage(gasCap.Gas(), header.BaseFee) 
+	if err != nil {
+		return nil, err
+	}
+	tx := args.ToTransaction(types.LegacyTxType)
+	if msg.GasPrice().Sign() == 0 {
+		header.BaseFee = new(big.Int)
+	}
+	var traceConfig *TraceConfig
+	if config != nil {
+		traceConfig = &config.TraceConfig
+	}
+	return s.traceTx(ctx, tx, msg, new(TracerContext), header, statedb, evmFn, traceConfig)
+}
+
+// traceTx configures a new tracer according to the provided configuration, and
+// executes the given message in the provided environment. The return value will
+// be tracer dependent.
+func (s *PrivateDebugAPI) traceTx(ctx *rpc.CallContext, tx *types.Transaction, message Msg, txctx *TracerContext, header *types.Header, statedb state.StateDB, getEVM func(state.StateDB, *vm.Config, common.Address, *big.Int) *vm.EVM, config *TraceConfig) (interface{}, error){
+	var ( 
+		timeout = defaultTraceTimeout
```

Comment:
I'd like to see a configuration option to let people set this higher at a service config level.